packages/variable-resolver/src/browser/variable-resolver-service.ts

```diff
@@ -116,17 +117,25 @@ export namespace VariableResolverService {
     export class Context {
 
         protected readonly resolved = new Map<string, string | undefined>();
+        protected readonly overridden = new Map<string, string | undefined>();
```

Comment:
Why cannot we just use `resolved`?

Reply:
because variables like ${cwd} could possibly be different from one task to another, e.g.,
```js
[
{ type: 'shell', label: 'task 1', cwd: '/home/user/theia', ... },
{ type: 'shell', label: 'task 2', cwd: '/home/user', ... }
]
```
if i use `resolved` and run 'task 1' and then 'task 2', ${cwd} would be interpreted as '/home/user/theia' for both tasks.